### ISSUE TITLE
Svenza Revel Robot on ROS Melodic Ubuntu 18.04

### ISSUE THREAD

**Issue by UofI-CDACS** [SvenzvaRobotics/svenzva_ros#26]
Max,

Something got screwed up in my ROS installation and I can no longer run the Revel robot.  The revel_dynamic_model dies when running the svenzva_bringup.launch

I would like to reintall the melodic branch, but I can't find it on your github repository.

What am I missing here?

John Shovic

**Comment by maxsvetlik** [comment]
Hey John,

As of PR #25 the master/main branch now targets melodic, and should
incorporate all the changes that were on the melodic branch you were using
in Issue #23.

Out of caution, I've restored the feature/melodic branch for you to compare
against.

Let me know how it works out

Cheers
Max

**Comment by UofI-CDACS** [comment]
Max,

I reloaded and compiled from the main branch.  Rebooted the computer,  power cycled the Arm, made sure it was turned on, etc.



Here are the errors I get:

robotoperator@rosbox3:~/catkin_ws$ roslaunch svenzva_drivers svenzva_bringup.launch
... logging to /home/robotoperator/.ros/log/f7e42460-237a-11eb-b127-00e18ca380ef/roslaunch-rosbox3-20527.log
Checking log directory for disk usage. This may take a while.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

started roslaunch server http://rosbox3:39927/

SUMMARY
========

PARAMETERS
 * /arm_speed_limit: 40.0
 * /collision_check_enabled: False
 * /joint_names: ['joint_1', 'join...
 * /joint_state_publisher/arm_namespace: revel
 * /joint_state_publisher/rate: 50
 * /max_motor_id: 7
 * /num_motors_present: 7
 * /revel_driver/arm_namespace: revel
 * /revel_driver/mode: user_defined
 * /revel_driver/param_file: control_params.yaml
 * /revel_driver/update_rate: 50
 * /robot_description: <?xml version="1....
 * /rosdistro: melodic
 * /rosversion: 1.14.9

NODES
  /
    joint_state_publisher (svenzva_drivers/joint_state_publisher.py)
    revel_driver (svenzva_drivers/svenzva_driver.py)
    revel_dynamic_model (svenzva_drivers/svenzva_dynamics)
    rob_st_pub (robot_state_publisher/robot_state_publisher)

ROS_MASTER_URI=http://localhost:11311

process[joint_state_publisher-1]: started with pid [20542]
process[rob_st_pub-2]: started with pid [20543]
process[revel_driver-3]: started with pid [20544]
process[revel_dynamic_model-4]: started with pid [20546]
[ INFO] [1605031682.504772434]: Loading model from /home/robotoperator/catkin_ws/src/svenzva_ros/svenzva_description/robots/svenzva_arm.urdf
[ INFO] [1605031682.507813229]: Kinematic chain expects 6 joints
[INFO] [1605031682.795526]: Starting Joint State Publisher at
started roslaunch server http://rosbox3:39927/ 50Hz
[ WARN] [1605031683.033453209]: Received JointState is 1605031683.033322 seconds old.
[INFO] [1605031684.354148]: revel: Pinging motor IDs 1 through 7...
[revel_dynamic_model-4] process has died [pid 20546, exit code -11, cmd /home/robotoperator/catkin_ws/devel/lib/svenzva_drivers/svenzva_dynamics __name:=revel_dynamic_model __log:=/home/robotoperator/.ros/log/f7e42460-237a-11eb-b127-00e18ca380ef/revel_dynamic_model-4.log].
log file: /home/robotoperator/.ros/log/f7e42460-237a-11eb-b127-00e18ca380ef/revel_dynamic_model-4*.log
^C[revel_driver-3] killing on exit
[rob_st_pub-2] killing on exit
[joint_state_publisher-1] killing on exit
[revel_driver-3] escalating to SIGTERM
[revel_driver-3] escalating to SIGKILL
Shutdown errors:
 * process[revel_driver-3, pid 20544]: required SIGKILL. May still be running.
shutting down processing monitor...
... shutting down processing monitor complete
done



Interestingly, after shutting down the roslaunch, I get this for rostopics and rosnode:



robotoperator@rosbox3:~/catkin_ws$ rostopic list
/joint_states
/revel/model_efforts
/revel/motor_states
/rosout
/rosout_agg
robotoperator@rosbox3:~/catkin_ws$ rosnode list
/revel_driver
/revel_dynamic_model
/rosout
robotoperator@rosbox3:~/catkin_ws$

**Comment by maxsvetlik** [comment]
[INFO] [1605031684.354148]: revel: Pinging motor IDs 1 through 7...

Indicates that the driver is trying to discover the motors.  The fact that
the dynamic_model module crashes indicates that the motors aren't being
found.

As in the previous issue, can you verify that the red led is flashing on
the motor in Joint 2 when the power is first restored?

**Comment by UofI-CDACS** [comment]
Yes, the light on joint2 blinks once when the power is turned on.

John
________________________________

**Comment by UofI-CDACS** [comment]
Yes, the light on joint2 blinks once when the power is turned on.

John

**Comment by UofI-CDACS** [comment]
Any ideas on this problem?

I have power cycled everything, made sure all the lights come on in the Robot arm, and I still get exactly the same result.

John

**Comment by maxsvetlik** [comment]
If the lights flash on the motors, that means they're all connected to power and the USB-to-serial device.
The only thing I can think to check would be the permissions of the USB device. What's the output from the following?

`$ ls /dev/ttyUSB0 -lah`

**Comment by UofI-CDACS** [comment]
Max,
crw-rw-rw- 1 root dialout 188,0 Nove 16 11:54 /dev/ttyUSB0

This does not appear to be the problem   The Arm is not responding to any commands.   Do you have a simple way of testing the robot outside of ROS?

open_gripper no longer works either.  I have totally rebooted all computers and the arm.  Same results.

I *believe* we now have a bad robot USB interface, but I'm not sure how to verify this.

John

**Comment by maxsvetlik** [comment]
I see, thanks for the output. The permissions for the USB file descriptor look OK. It could indeed be a malfunction of the USB interface (though the fact that it is detected as a USB device on your machine is a good sign)

The easiest way to debug connectivity to the motors outside of ROS would be to use the motor manufacture's motor manager software. You can find its manual [here](https://emanual.robotis.com/docs/en/software/rplus2/manager/) and is a Windows-only program I believe. Note that you will want to install the R+ 2.0 software, and will select `MX-64` from the motor selection upon running.
